src/main.rs

```diff
@@ -112,8 +123,8 @@ struct GetSignalsQ {
     url: String,
 }
 
-async fn get_signals(uid: i64, q: GetSignalsQ, pool: DB) -> eyre::Result<Signals> {
-    Signals::get(uid, q.url, &pool)
+async fn get_signals(account: AccountSession, q: GetSignalsQ, pool: DB) -> eyre::Result<Signals> {
```

Comment:
The method doesn't need anything except the uid, why do we want it to require an `AccountSession` parameter? Same for the other similar changes below.

Reply:
There's some argument that having a real type that can only be constructed by the function that validates the session is potentially safer, but the real reason is because it was simpler to have a single `authenticate` function that collected all the information used by any handler in a single query. Thought about having extra helpers/mapping after the fact to get back to just an `i64`, but they seemed equivalent to me.

Happy to add in an extra map to get back to `i64` if you prefer it.